src/GameState/index.tsx

```diff
@@ -0,0 +1,238 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import {
+  ArrowBackIcon,
+  Box,
+  Center,
+  CheckIcon,
+  Flex,
+  Heading,
+  HStack,
+  ScrollView,
+  SmallCloseIcon,
+  Spinner,
+  Stack,
+  Text,
+  View,
+} from 'native-base';
+import React, { useEffect, useReducer } from 'react';
+import { RootStackParamList } from '../../App';
+import Constants from 'expo-constants';
+import { Game, User } from '../interface';
+import { Board } from './Board';
+import { Marble } from './Marble';
+import { background } from 'styled-system';
+
+type Props = NativeStackScreenProps<RootStackParamList, 'GameState'>;
+const API_URL = Constants?.manifest?.extra?.API_URL;
+
+const initialState: {
+  players: User[];
+  game: Game | undefined;
+  isLoading: boolean;
+  error: any | undefined;
+} = {
+  players: [],
+  game: undefined,
+  isLoading: true,
+  error: undefined,
+};
+
+const reducer = (
+  state: {
+    players: User[];
+    game: Game | undefined;
+    isLoading: boolean;
+    error: any | undefined;
+  },
+  action: { type: string; value: any },
+) => {
+  return { ...state, [action.type]: action.value };
```

Comment:
I know it's more verbose, but I like it when action types are explicit. Here, I understand the logic of `type = "game"` to directly modify your state. 
But first of all, if your reducer gets complicated, you'll have to keep some of the logic handling the new values outside the reducer.
And furthermore, I find it much less readable. For example, a `type = "gameInitialization"` seems to me to make the code easier to understand.